# Copyright (c) 2006, 2012, Oracle and/or its affiliates. All rights reserved.
#
# Permission is hereby granted, free of charge, to any person obtaining a
# copy of this software and associated documentation files (the "Software"),
# to deal in the Software without restriction, including without limitation
# the rights to use, copy, modify, merge, publish, distribute, sublicense,
# and/or sell copies of the Software, and to permit persons to whom the
# Software is furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice (including the next
# paragraph) shall be included in all copies or substantial portions of the
# Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
# DEALINGS IN THE SOFTWARE.
6385111: Xorg auto-configuration Improvement

diff --git a/hw/xfree86/common/xf86Config.c b/hw/xfree86/common/xf86Config.c
index 891623d..ca1b1aa 100644
--- a/hw/xfree86/common/xf86Config.c
+++ b/hw/xfree86/common/xf86Config.c
@@ -1111,6 +1111,7 @@
 freeDevice(InputInfoPtr * list, InputInfoPtr pInfo)
 {
     InputInfoPtr *devs;
+    char **end, **drv, **drivers, **x;
 
     for (devs = list; devs && *devs; devs++) {
         if (*devs == pInfo) {

@@ -1993,7 +1993,7 @@ configMonitor(MonPtr monitorp, XF86ConfMonitorPtr conf_monitor)
     cmodep = conf_monitor->mon_modeline_lst;
     while (cmodep) {
         mode = xnfcalloc(1, sizeof(DisplayModeRec));
-        mode->type = 0;
+        mode->type = M_T_USERDEF;
         mode->Clock = cmodep->ml_clock;
         mode->HDisplay = cmodep->ml_hdisplay;
         mode->HSyncStart = cmodep->ml_hsyncstart;
